contracts/contracts/token/OUSD.sol

```diff
@@ -244,36 +327,92 @@ contract OUSD is Initializable, InitializableToken, Governable {
     function _burn(address _account, uint256 _amount) internal {
         require(_account != address(0), "Burn from the zero address");
 
-        _totalSupply = _totalSupply.sub(_amount);
-        uint256 creditAmount = _removeCredits(_account, _amount);
+        bool isNonRebasingAccount = _isNonRebasingAddress(_account);
+
+        uint256 creditAmount = _amount.mulTruncate(_creditsPerToken(_account));
+        uint256 currentCredits = _creditBalances[_account];
+
+        // Remove the credits
```

Comment:
nit: add comment to explain the currentCredit -1 trick. I guess it's for handling round error?